contracts/multisigs/PolySafeCreatorWithRecoveryModule.sol

```diff
@@ -0,0 +1,207 @@
+    /// @param _recoveryModule Recovery Module address.
+    constructor(address _polySafeProxyFactory, address _recoveryModule) {
+        if (_polySafeProxyFactory == address(0) || _recoveryModule == address(0)) {
+            revert ZeroAddress();
+        }
+
+        polySafeProxyFactory = _polySafeProxyFactory;
+        recoveryModule = _recoveryModule;
+    }
+
+    /// @dev Creates Poly Safe multisig.
+    /// @notice Number of owners and threshold is required to be 1.
+    /// @param owners Set of multisig owners.
+    /// @param threshold Number of required confirmations for a multisig transaction.
+    /// @param data Packed data related to the creation of a chosen multisig.
+    /// @return multisig Address of a created multisig.
+    function create(address[] memory owners, uint256 threshold, bytes memory data) external returns (address multisig) {
+        // PolySafe is created based on one owner only
+        if (owners.length != 1) {
+            revert WrongNumOwners(1, owners.length);
+        }
+
+        // Decode provided data
+        (IPolySafeProxyFactory.Sig memory safeCreateSig, bytes memory enableModuleSig) =
+            abi.decode(data, (IPolySafeProxyFactory.Sig, bytes));
+
+        // Calculate multisig address
+        multisig = IPolySafeProxyFactory(polySafeProxyFactory).computeProxyAddress(owners[0]);
+
+        // Check that multisig is not yet deployed
+        if (multisig.code.length > 0) {
+            revert MultisigAlreadyExists(multisig);
+        }
+
+        // Create a poly safe multisig via its proxy factory with all payment related values equal to zero
+        IPolySafeProxyFactory(polySafeProxyFactory).createProxy(address(0), 0, payable(address(0)), safeCreateSig);
+
+        // Check for zero address
+        if (multisig == address(0)) {
+            revert ZeroAddress();
+        }
+
+        // Get provided proxy multisig owners and threshold
+        address[] memory checkOwners = ISafe(multisig).getOwners();
+        uint256 checkThreshold = ISafe(multisig).getThreshold();
+
+        // Verify multisig proxy for provided owners and threshold
+        if (threshold != checkThreshold) {
+            revert WrongThreshold(checkThreshold, threshold);
+        }
+        uint256 numOwners = owners.length;
+        if (numOwners != checkOwners.length) {
+            revert WrongNumOwners(checkOwners.length, numOwners);
+        }
+
+        // Check for owner address match
+        if (owners[0] != checkOwners[0]) {
+            revert WrongOwner(owners[0]);
+        }
+
+        // Enable module payload
+        bytes memory execData = abi.encodeCall(ISafe.enableModule, (recoveryModule));
+
+        // Enable Recovery Module
+        // to = multisig, value = 0, operation = Call, all payment related = 0
+        ISafe(multisig)
+            .execTransaction(
+                multisig, 0, execData, ISafe.Operation.Call, 0, 0, 0, address(0), payable(address(0)), enableModuleSig
+            );
```

Comment:
`enableModule()` encoding and call. The signature must be provided along with polySafe creation signature